ganttproject/src/net/sourceforge/ganttproject/task/algorithm/RecalculateTaskCompletionPercentageAlgorithm.java

```diff
@@ -29,7 +29,7 @@ public void run() {
     }
     TaskContainmentHierarchyFacade facade = createContainmentFacade();
     for (Task t : facade.getTasksInDocumentOrder()) {
-      if (!facade.hasNestedTasks(t)) {
+      if (facade.getDepth(t) == 0) {
```

Comment:
Since the algorithm is now going top-down, you can remove this loop completely and just take the root task from the facade.

Reply:
Ah, ok. So there is always a single root task no matter of the project structure?